src/Hl7.Fhir.Specification/Specification/Source/CommonZipSource.cs

```diff
@@ -34,30 +37,35 @@ public class CommonZipSource : ISummarySource, ICommonConformanceSource, IArtifa
         /// <summary>Create a new <see cref="CommonZipSource"/> instance for the ZIP archive with the specified file path.</summary>
         /// <param name="inspector"></param>
         /// <param name="zipPath">File path to a Zip archive.</param>
-        /// <param name="targetDirectory">The directory within the users temp directory to unzip the archive to.</param>
-        public CommonZipSource(ModelInspector inspector, string zipPath, string targetDirectory) : this(inspector, zipPath, targetDirectory, DirectorySourceSettings.CreateDefault()) { }
-
-        /// <summary>Create a new <see cref="CommonZipSource"/> instance for the ZIP archive with the specified file path.</summary>
-        /// <param name="inspector"></param>
-        /// <param name="zipPath">File path to a ZIP archive.</param>
-        /// <param name="targetDirectory">The directory within the users temp directory to unzip the archive to.</param>
+        /// <param name="cacheDirectory">Path of the directory to unzip the archive to.</param>
         /// <param name="settings">Configuration settings for the internal <see cref="CommonDirectorySource"/> instance.</param>
-        public CommonZipSource(ModelInspector inspector, string zipPath, string targetDirectory, DirectorySourceSettings settings)
+        public CommonZipSource(ModelInspector inspector, string zipPath, string cacheDirectory, DirectorySourceSettings settings)
         {
-            if (string.IsNullOrEmpty(zipPath)) { throw Error.ArgumentNull(nameof(zipPath)); }
-            if (string.IsNullOrEmpty(targetDirectory)) { throw Error.ArgumentNull(nameof(targetDirectory)); }
+            if (string.IsNullOrEmpty(zipPath)) throw Error.ArgumentNull(nameof(zipPath));
+            if (string.IsNullOrEmpty(cacheDirectory)) throw Error.ArgumentNull(nameof(cacheDirectory));
+            if (settings is null) throw Error.ArgumentNull(nameof(settings));
+
             _inspector = inspector;
             ZipPath = zipPath;
-            TargetDirectory = targetDirectory;
-            if (settings == null) { throw Error.ArgumentNull(nameof(settings)); }
+            CacheDirectory = cacheDirectory;
+
             // Always clone the incoming reference, especially since we're forcing IncludeSubDirectories
             _settings = settings.Clone();
             _lazySource = new Lazy<CommonDirectorySource>(createSource);
+            _mask = settings.Mask;
         }
 
+        /// <inheritdoc cref="CommonZipSource.CommonZipSource(ModelInspector, string, string, DirectorySourceSettings)"/>
+        public CommonZipSource(ModelInspector inspector, string zipPath, string cacheDirectory)
+            : this(inspector, zipPath, cacheDirectory, DirectorySourceSettings.CreateDefault()) { }
+
         /// <summary>Gets the location of the ZIP archive, as specified in the constructor.</summary>
         public string ZipPath { get; }
-        public string TargetDirectory { get; }
```

Comment:
This is a breaking change that we have to document.

Reply:
Wasn't CommonZipSource new? So there is no current public version of this class?  I'll double check...

Reply:
I double checked, TargetDirectory was introduced by me in a previous PR (for 5.x), so it does not exist on the currently public ZIpSource/CommonZipSource yet, so no breaking change to document.